Samples/Notifications/Push/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,161 @@
+                std::cout << "\nWNS Channel URI request is in progress." << std::endl;
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The WNS Channel URI request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result{ co_await channelOperation };
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channel{ result.Channel() };
+
+        std::cout << "\nWNS Channel URI: " << winrt::to_string(channel.Uri().ToString()) << std::endl;
+
+        // It's the caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with the WNS Channel URI request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task{ RequestChannelAsync() };
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result{ task.GetResults() };
+    return result;
+}
+
+// Subscribe to an event which will get signaled whenever a foreground notification arrives.
+void SubscribeForegroundEventHandler()
+{
+    winrt::event_token token{ PushNotificationManager::Default().PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+    {
+        auto payload{ args.Payload() };
+
+        std::string payloadString(payload.begin(), payload.end());
+        std::cout << "\nPush notification content received in the FOREGROUND: " << payloadString << std::endl;
+    }) };
+}
+
+int main()
+{
+    // Setup an event handler, so we can receive notifications in the foreground while the app is running.
+    SubscribeForegroundEventHandler();
+
+    PushNotificationManager::Default().Register();
+
+    auto args{ AppInstance::GetCurrent().GetActivatedEventArgs() };
+    auto kind{ args.Kind() };
+    switch (kind)
+    {
+        // When it is launched normally (by the users, or from the debugger), the sample requests a WNS Channel URI and
+        // displays it, then waits for notifications. This user can take a copy of the WNS Channel URI and use it to send
+        // notifications to the sample
+        case ExtendedActivationKind::Launch:
+        {
+            // Request a WNS Channel URI which can be passed off to an external app to send notifications to.
```

Comment:
Is there a plan to add the IsSupported check here once preview 2 is out?

`            // Check if push notifications are supported. Certain self-contained apps may not support push notifications by design
            if (PushNotificationManager::IsSupported())
            {
                // Request a WNS Channel URI which can be passed off to an external app to send notifications to.
                // The WNS Channel URI uniquely identifies this app for this user and device.
                PushNotificationChannel channel{ RequestChannel() }; 
`

Reply:
We definitely should :)

Reply:
there is a plan, but not for preview1

Reply:
config/gni/devtools_grd_files.gni
front_end/core/platform/BUILD.gn
front_end/core/platform/UserVisibleError.ts
front_end/core/platform/platform.ts
front_end/panels/elements/ElementsPanel.ts